solidity/contracts/token/TokenBridgeCctpV1.sol

```diff
@@ -100,68 +74,48 @@ contract TokenBridgeCctpV1 is TokenBridgeCctpBase, IMessageHandler {
     function _validateHookMessage(
         bytes calldata hyperlaneMessage,
         bytes29 cctpMessage
-    ) internal view override {
-        bytes32 circleSender = cctpMessage._sender();
-        require(
-            circleSender == _mustHaveRemoteRouter(hyperlaneMessage.origin()),
-            "Invalid circle sender"
-        );
-
+    ) internal pure override {
         bytes32 circleMessageId = cctpMessage._messageBody().index(0, 32);
         require(circleMessageId == hyperlaneMessage.id(), "Invalid message id");
     }
 
     /// @inheritdoc IMessageHandler
     function handleReceiveMessage(
-        uint32 /*sourceDomain*/,
-        bytes32 /*sender*/,
-        bytes calldata /*body*/
-    ) external pure override returns (bool) {
+        uint32 sourceDomain,
+        bytes32 sender,
+        bytes calldata body
+    ) external override returns (bool) {
+        _authenticateCircleSender(sourceDomain, sender);
+        preVerifyMessage(_messageId(body), 0);
         return true;
     }
 
+    function _messageId(bytes calldata body) internal pure returns (bytes32) {
```

Comment:
This seems different of how we typically access destructured info in bodies via a library. I'd say either do that, or use a more descriptive function name